gammapy/background/cube.py

```diff
@@ -809,6 +809,38 @@ def plot_spectrum(self, coord, ax=None, style_kwargs=None):
             plt.close(fig)
         return ax
 
+    @property
+    def integral(self):
+        """Integral of the cube (`~astropy.units.Quantity`)"""
+        delta_energy = self.energy_edges[1:] - self.energy_edges[:-1]
+        delta_y = self.coordy_edges[1:] - self.coordy_edges[:-1]
+        delta_x = self.coordx_edges[1:] - self.coordx_edges[:-1]
+        # define grid of deltas (i.e. bin widths for each 3D bin)
+        delta_energy, delta_y, delta_x = np.meshgrid(delta_energy, delta_y,
+                                                     delta_x, indexing='ij')
+        bin_volume = delta_energy*(delta_y*delta_x).to('sr')
+        integral = self.data*bin_volume
+
+        return integral.sum()
+
+    @property
+    def integral_images(self):
+        """Integral of the cube images (`~astropy.units.Quantity`)
+
+        Calculate the integral of each energy bin (slice) in the
+        cube. Returns an array of integrals.
```

Comment:
Again, from the method name and docstring it's not obvious to me what this does.

It looks to me like this is computing a 1-dim `npred` array for each energy bin?